src/catalog/user/user.controller.js

```diff
@@ -119,3 +120,42 @@ export const findAllAccountsOfUser = async (req, res, next) => {
     })
   }
 }
+
+export const generateAndSendOTP = async (req, res, next) => {
+  try {
+    const {id, name, email} = req.user
+    const userID = id
+
+    const otpDigit = generators.generateOTPDigit() // Generate otpDigits
+    
+    // Find old OTP of user and remove from the DB
+    const oldOTP = await otpService.findOTPByUserID(userID)
+
+    if (oldOTP) {
+      await oldOTP.destroy()
+    }
+
+    // Save new OTP to DB
+    await otpService.createOTP(userID, otpDigit)
+
+    //  Create template for otp email
+    const html = generators.generateHTMLEmail(name, otpDigit)
+
+    await userService.sendOTPMail(email, html) // service send email OTP to user email
+
+    //for testing -> replace your email to receive OTP from the server, then you can check your mail box
+    //for example: await userService.sendOTPMail('hoangnghia.binhthuan@gmail.com', html)
+
+    return res.status(httpStatusCodes.OK).json({
+      message: MESSAGE.OK,
+      payload: req.user,
+      msg: `OTP already send to email: ${email}`
```

Comment:
có message: MESSAGE.OK rồi thì ko cần thêm msg: `OTP already send to email` đâu, và t thấy cũng ko nhất thiết trả thêm req.user làm gì, trả nhiều thông tin ko dùng lại làm nặng response, khi nào ai code frontend cái này cần gì tự thêm vào

Reply:
Ok